### PR TITLE
[BUG] Certification test 5.7.1 fails

### PR DESCRIPTION
**Pull request by tomereli**
Controller certification tests 5.7.1 fails on fefec5f.

Logs can be found here: https://gitlab.com/prpl-foundation/prplMesh/-/jobs/401365002/artifacts/browse/logs/FAIL/

### CODE REVIEW

**Comment by rmelotte** [comment]
Rebased on master after https://github.com/prplfoundation/prplMesh/pull/766 was merged.

File: framework/tlvf/src/src/CmduMessage.cpp [hunk 1]
```diff
@@ -65,3 +65,12 @@ uint16_t CmduMessage::getMessageId()
 
     return mid;
 }
+
+void CmduMessage::setMessageId(uint16_t mid)
+{
+    auto cmduhdr = getCmduHeader();
+
+    if (cmduhdr->is_finalized())
+        swap_16((uint16_t &)mid);
```
Comment (arnout) [inline framework/tlvf/src/src/CmduMessage.cpp]: This cast looks fishy. Why is it needed?

File: agent/src/beerocks/slave/son_slave_thread.cpp [hunk 2]
```diff
@@ -4384,38 +4384,16 @@ bool slave_thread::handle_client_steering_request(Socket *sd, ieee1905_1::CmduMe
...
-            if (!ap_metrics_query_tlv) {
-                LOG(ERROR) << "addClass tlvApMetricQuery has failed";
-                return false;
-            }
-            LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE, mid=" << std::hex << int(mid)
-                       << ", with TLV type=" << std::hex << int(ap_metrics_query_tlv->type());
-        }
-        tlv_type = cmdu_tx.getNextTlvType();
-    }
+    LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE , mid=" << std::hex << int(mid);
```
Comment (vitalybu) [inline agent/src/beerocks/slave/son_slave_thread.cpp]:
```suggestion
    LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE, mid=" << std::hex << int(mid);
```

File: agent/src/beerocks/slave/son_slave_thread.cpp [hunk 3]
```diff
@@ -4384,38 +4384,16 @@ bool slave_thread::handle_client_steering_request(Socket *sd, ieee1905_1::CmduMe
...
-            if (!link_metrics_query_tlv) {
-                LOG(ERROR) << "addClass tlvLinkMetricQuery has failed";
-                return false;
-            }
-            LOG(DEBUG) << "Received LINK_METRIC_QUERY_MESSAGE , mid=" << std::hex << int(mid)
-                       << ", with TLV type=" << std::hex << int(link_metrics_query_tlv->type());
-        }
-        tlv_type = cmdu_tx.getNextTlvType();
-    }
+    LOG(DEBUG) << "Received LINK_METRIC_QUERY_MESSAGE , mid=" << std::hex << int(mid);
```
Comment (vitalybu) [inline agent/src/beerocks/slave/son_slave_thread.cpp]:
```suggestion
    LOG(DEBUG) << "Received LINK_METRIC_QUERY_MESSAGE, mid=" << std::hex << int(mid);
```

**Review by vitalybu — APPROVED** [review]

File: framework/tlvf/src/src/CmduMessage.cpp (hunk 1 above)
Comment (rmelotte) [inline framework/tlvf/src/src/CmduMessage.cpp]: Is that a c-style cast (I know there are others in the file...)
~~Also, if you make setMessageId accept a reference instead, you don't need any cast.~~
Edit: the cast is not needed even if setMessageId doesn't take mid by reference. Does the caller of setMessageId need `mid` to be swapped?

**Review by rmelotte — COMMENTED** [review]
Please remove the c-style cast by any mean.
Otherwise, LGTM!

File: framework/tlvf/src/src/CmduMessage.cpp (hunk 1 above)
Comment (tomereli) [reply framework/tlvf/src/src/CmduMessage.cpp]: Copy-paste https://tse1.mm.bing.net/th?id=OGC.335506d103b360a3e59d56ddfa94fdb8&pid=Api&rurl=https%3a%2f%2fmedia.giphy.com%2fmedia%2f26xBN67Q1HzBDyU0w%2fsource.gif&ehk=1ypHjLI5Si28m2%2fbCzFpq6QelYRKn21j2RQUPFdV4wo%3d

Comment (tomereli) [reply framework/tlvf/src/src/CmduMessage.cpp]: I fixed it and got rid of the other C style casts in this file.

**Review by rmelotte — COMMENTED** [review]
(Github insists on requiring a comment to mark approval)

**Review by YuriyMasechko — APPROVED** [review]

**Review by rmelotte — COMMENTED** [review]
[]

**Pull request merged**